File: assets/scripts/moveduck.py
import dengine.common as common
import dengine.scene as scene
import dengine.inpt as inpt
import dengine.timer as timer
import dengine.filesys as filesys
speed = 4.0


def start(entity):
    print("Start invoked from C to Python")
    print("files:",filesys.get_filesdir())

def update(entity):
    delta_s = timer.get_delta() / 1000.0
    if(inpt.get_key('A')):
        entity.transform.position.x -= delta_s * speed
    if(inpt.get_key('D')):
        entity.transform.position.x += delta_s * speed
    if(inpt.get_key('E')):
        entity.transform.position.y += delta_s * speed
    if(inpt.get_key('C')):
        entity.transform.position.y -= delta_s * speed
    if(inpt.get_key('W')):
        entity.transform.position.z -= delta_s * speed
    if(inpt.get_key('S')):
        entity.transform.position.z += delta_s * speed
    if(inpt.get_key('Z')):
        entity.transform.rotation.y += delta_s * speed * 30.0
    if(inpt.get_key('X')):
        entity.transform.rotation.y -= delta_s * speed * 30.0

File: nsl/src/nsl-test/moveduck.c
#include <dengine/dengine.h>

int moveduck_start(Entity* entity)
{
    printf("Start invoked from C to NSL\n");
    printf("files: %s\n", dengineutils_filesys_get_filesdir());
    return 0;
}

int moveduck_update(Entity* entity)
{
    static const double speed = 4.0f;
    double delta_s = dengineutils_timer_get_delta() / 1000.0;
    if(dengine_input_get_key('A'))
       entity->transform.position[0] -= delta_s * speed;
    if(dengine_input_get_key('D'))
       entity->transform.position[0] += delta_s * speed;
    if(dengine_input_get_key('E'))
       entity->transform.position[1] += delta_s * speed;
    if(dengine_input_get_key('C'))
       entity->transform.position[1] -= delta_s * speed;
    if(dengine_input_get_key('W'))
       entity->transform.position[2] -= delta_s * speed;
    if(dengine_input_get_key('S'))
       entity->transform.position[2] += delta_s * speed;
    if(dengine_input_get_key('Z'))
       entity->transform.rotation[1] += delta_s * speed * 30.0;
    if(dengine_input_get_key('X'))
       entity->transform.rotation[1] -= delta_s * speed * 30.0;

    return 0;
}
